src/checkpoint/checkpoint.h

```diff
@@ -81,6 +81,97 @@ struct __wt_ckpt_connection {
     uint64_t last_base_write_gen;
 };
 
+/*
+ * Block modifications from an incremental identifier going forward.
+ */
+/*
+ * At the default granularity, this is enough for blocks in a 2G file.
+ */
+#define WT_BLOCK_MODS_LIST_MIN 128 /* Initial bits for bitmap. */
+struct __wt_block_mods {
+    const char *id_str;
+
+    WT_ITEM bitstring;
+    uint64_t nbits; /* Number of bits in bitstring */
+
+    uint64_t offset; /* Zero bit offset for bitstring */
+    uint64_t granularity;
+/* AUTOMATIC FLAG VALUE GENERATION START 0 */
+#define WT_BLOCK_MODS_RENAME 0x1u /* Entry is from a rename */
+#define WT_BLOCK_MODS_VALID 0x2u  /* Entry is valid */
+                                  /* AUTOMATIC FLAG VALUE GENERATION STOP 32 */
+    uint32_t flags;
+};
+
+/*
+ * WT_CKPT --
+ *	Encapsulation of checkpoint information, shared by the metadata, the
+ * btree engine, and the block manager.
+ */
+#define WT_CHECKPOINT "WiredTigerCheckpoint"
+#define WT_CKPT_FOREACH(ckptbase, ckpt) for ((ckpt) = (ckptbase); (ckpt)->name != NULL; ++(ckpt))
+#define WT_CKPT_FOREACH_NAME_OR_ORDER(ckptbase, ckpt) \
+    for ((ckpt) = (ckptbase); (ckpt)->name != NULL || (ckpt)->order != 0; ++(ckpt))
+
+struct __wt_ckpt {
+    char *name; /* Name or NULL */
+
+    /*
+     * Each internal checkpoint name is appended with a generation to make it a unique name. We're
+     * solving two problems: when two checkpoints are taken quickly, the timer may not be unique
+     * and/or we can even see time travel on the second checkpoint if we snapshot the time
+     * in-between nanoseconds rolling over. Second, if we reset the generational counter when new
+     * checkpoints arrive, we could logically re-create specific checkpoints, racing with cursors
+     * open on those checkpoints. I can't think of any way to return incorrect results by racing
+     * with those cursors, but it's simpler not to worry about it.
+     */
+    int64_t order; /* Checkpoint order */
+
+    uint64_t sec; /* Wall clock time */
+
+    uint64_t size; /* Checkpoint size */
+
+    uint64_t write_gen;     /* Write generation */
+    uint64_t run_write_gen; /* Runtime write generation. */
+
+    char *block_metadata;   /* Block-stored metadata */
+    char *block_checkpoint; /* Block-stored checkpoint */
+
+#define WT_BLKINCR_MAX 2
```

Comment:
I agree with @etienneptl 's earlier comment that moving this into the structure doesn't feel right. There is a different backup related structure that also uses this maximum value as well as a lot of loops in backup code. Maybe this should be in `connection.h` as a more "global" place to put it.

Reply:
I don't think I can put it into `connection.h` because of dependencies. The more "global" thing depends on more "local" ones in this case.

Reply:
Is it ok to move it to `block.h"?

Reply:
Or better `cursor.h`?

Reply:
I moved it to `cursor.h` because it's a part of backup. We don't have a "backup" module but we do have a backup cursor, so that's the closest to that.